src/wristpy/processing/calibration.py

```diff
@@ -0,0 +1,359 @@
+"""Calibrate accelerometer data."""
+
+import math
+import typing
+
+import numpy as np
+from sklearn import linear_model
+
+from wristpy.core import computations, models
+
+
+class SphereCriteriaError(Exception):
+    """Raises error when calibration fails due to sphere criteria."""
+
+    def __init__(self) -> None:
+        """Initalize."""
+        self.message = "Calibration failed, data did not meet sphere criteria."
+        super().__init__(self.message)
+
+
+class CalibrationError(Exception):
+    """Raises error when calibration error can not get below error tolerance."""
+
+    def __init__(self) -> None:
+        """Init."""
+        self.message = "Calibration failed, error could not be sufficiently improved"
+        super().__init__(self.message)
+
+
+class Calibration:
+    """Implements calibration on accelerometer data, based off of GGIR's implementation.
+
+    This class implements methods for autocalibrating accelerometer data using either
+    entire dataset or subsets, as determined by the settings. Depending on the settings
+    a scale and offset value is determined and applied to the data which, if
+    successful, is returned to the user.
+
+    Attributes:
```

Comment:
Just going over the style guide again. I think we only define the class Attributes once? Either here or in __init__, not both? 
@ReinderVosDeWael

Reply:
Good question I know it came up in your review as well, I changed things around so that there's a brief discussion as attributes after the class description and more detailed ones in the __init__. What are best practices ?